README.md

```diff
@@ -1,122 +1,241 @@
+map the URI of a given public RRDP notification URI to an alternative hidden backend server:
 
-Other means of synchronizing the content (e.g. rsync, scp, or possibly even file system replication tools like GlusterFS) are not RRDP aware and thus may sync the Notification File before the files it references have all been downloaded which can cause Relying Party clients of the public facing repository servers to encounter errors while attempting to download updates from the service.
+```
+# krill-sync --source_uri_base https://hidden.server/rrdp/ https://some.server/rrdp/notification.xml
+```
 
-By default directory content for serving by an Rsync daemon is also created from the data downloaded via RRDP, no `rsync` binary is needed.
+There is a further assumption that snapshot and delta file URIs will use the same hostname
+and base path as the notification file, so that it can be resolved to the back-end using the
+same `--source_uri_base` map that is used for the notification file itself.
 
-## Usage
+Using the RRDP protocol as a source ensures that only complete and consistent data is downloaded
+by following the Notification File to the dependent resources (assuming a publication server
+such as Krill that ensures that the Notification File is written last during a content
+update). It also enables the public servers to download only deltas rather than the entire
+snapshot on update keeping the sync time, bandwidth usage and origin server impact to a minimum.
 
-This tool is intended to be used as a system "service" invoked periodically e.g. once a minute from cron, it is not meant to be a user facing tool. As such it follows [Linux FHS](https://refspecs.linuxfoundation.org/fhs.shtml) guidelines for storing application state rather than user state.
+Other means of synchronizing the content (e.g. rsync, scp, or possibly even file system
+replication tools like GlusterFS) are not RRDP aware and thus may sync the Notification File
+before the files it references have all been downloaded which can cause Relying Party clients
+of the public facing repository servers to encounter errors while attempting to download updates
+from the service.
+
+## Synchronization
 
-The simplest use case (assuming that the invoking user has write access to the default directories, see below) requires only the URL where the RFC 8182 RRDP Notification File can be downloaded, e.g.:
+When `krill-sync` runs without previous state, it will retrieve and parse the specified
+notification file. Then it will retrieve and parse the snapshot file to obtain all current
+objects, as well as each delta file mentioned (unless an optional hard limit is set, see below).
 
+If a file cannot be retrieved or **parsed** then the run will fail. This means that if the
+back-end system RFC 8182 RRDP XML is somehow not understood by krill-sync the data will
+not be accepted. We believe that this should not happen because of our testing, so this should
```

Comment:
Maybe drop the "We believe" sentence? Also move the "If a file cannot be retrieved" sentence below "On subsequen runs" as it applies to both initial and subsequent runs?

Reply:
ack